src/main/lpfConverter.ts

```diff
@@ -0,0 +1,304 @@
+                        link.Href = url;
+                    }
+                }
+                {
+                    const encodingFormat = ressource.encodingFormat;
+                    if (typeof encodingFormat === "string") {
+                        link.TypeLink = encodingFormat;
+
+                    } else {
+                        const ext = extname(link.Href);
+                        switch (ext) {
+                            case ".mp3":
+                                link.TypeLink = "audio/mpeg";
+                                break;
+                            case ".wav":
+                                link.TypeLink = "audio/wav";
+                                break;
+                            case ".opus":
+                                link.TypeLink = "audio/ogg";
+                                break;
+                            default:
+                                link.TypeLink = ""; // typeLink required
+                        }
+                    }
+                }
+                {
+                    const raw = ressource.name;
+                    const name = Array.isArray(raw) ? raw[0] : raw;
+                    if (typeof name === "string") {
+                        link.Title = name;
+                    }
+                }
+                {
+                    const raw = ressource.rel;
+                    const rawArray = Array.isArray(raw) ? raw : [raw];
+                    const rel = rawArray.filter((_rel) => typeof _rel === "string");
+                    if (rel.length) {
+                        link.Rel = rel;
+                    }
+                }
+                {
+                    const iso8601 = `${ressource.duration || ""}`;
+                    const second = iso8601DurationsToSeconds(iso8601);
+                    if (second > 0) {
+                        link.Duration = second;
+                    }
+                }
+                {
+                    const alternate = ressource.alternate;
+                    const children = convertW3CpublicationLinksToReadiumManifestLink(alternate);
+                    if (children.length) {
+                        link.Children = children;
+                    }
+                }
+
+                linkArray.push(link);
+            });
+
+    } else {
+        debug("no resources found");
+    }
+
+    return linkArray;
+}
+
+export async function lpfToAudiobookConverter(lpfPath: string): Promise<string> {
+
+    const lpfRenameInAudiobookPath = await copyAndRenameLpfFile(lpfPath);
+
+    const stream = await openAndExtractPublicationFromLpf(lpfRenameInAudiobookPath);
+
+    const buffer = await streamToBufferPromise(stream);
+    const rawData = buffer.toString("utf8");
+    const w3cManifest = JSON.parse(rawData) as JsonMap;
+
+    const readiumManifest = w3cPublicationManifestToReadiumPublicationManifest(w3cManifest);
+
+    const manifestBuffer = Buffer.from(JSON.stringify(readiumManifest));
```

Comment:
I would suggest formatting the JSON output:
`JSON.stringify(readiumManifest, null, 4)`

Reply:
done :)